core.cc

```diff
@@ -0,0 +1,181 @@
+#include <node.h>
+#include <nan.h>
+#include <syslog.h>
+
+using node::Buffer;
+
+using v8::Function;
+using v8::FunctionTemplate;
+using v8::Local;
+using v8::Null;
+using v8::Number;
+using v8::Object;
+using v8::String;
+using v8::Value;
+
+namespace {
+
+class Worker: public NanAsyncWorker {
+    public:
+        Worker(NanCallback *callback, int priority, char* message)
+            : NanAsyncWorker(callback), priority(priority), message(message) {
+            }
+        ~Worker() {
+            delete[] message;
+        }
+
+        void Execute() {
+            syslog(priority, "%s", message);
+        }
+
+        void HandleOKCallback() {
+            NanScope();
+
+            if(!callback->IsEmpty())
+                callback->Call(0, NULL);
+        };
+
+    private:
+        int priority;
+        char* message;
+};
+
+static char ident[1024];
+
+// void openlog(const char *ident, int option, int facility);
+NAN_METHOD(OpenLog) {
+    NanScope();
+
+    // openlog requires ident be statically allocated. Write doesn't guarantee
+    // NULL-termination, so preserve last byte as NULL.
+    args[0]->ToString()->WriteUtf8(ident, sizeof(ident)-1);
+    int option = args[1]->ToInt32()->Value();
+    int facility = args[2]->ToInt32()->Value();
+
+    openlog(ident, option, facility);
+
+    NanReturnUndefined();
+}
+
+static char* dupBuf(const char* mem, size_t memsz) {
+    char* s = new char[memsz + 1];
+    memcpy(s, mem, memsz);
+    s[memsz] = 0;
+    return s;
+}
+
+static char* dupStr(const Local<String>& m) {
+    if(m.IsEmpty())
+        return NULL;
+
+    // Exact calculation of UTF length involves double traversal. Avoid
+    // this by because we know UTF8 expansion is < 4 bytes out per byte
+    // in.
+    char* s = new char[m->Length() * 4];
+    m->WriteUtf8(s);
+    return s;
+}
+
+// void syslog(int priority, const char *format, ...);
+NAN_METHOD(SysLog) {
+    NanScope();
+
+    int priority = args[0]->ToInt32()->Value();
+    char* message = NULL;
+    NanCallback *callback = new NanCallback(args[2].As<Function>());
+
+    if(Buffer::HasInstance(args[1])) {
+        message = dupBuf(Buffer::Data(args[1]), Buffer::Length(args[1]));
+    } else {
+        message = dupStr(args[1]->ToString());
+    }
+
+    if (message) {
+        NanAsyncQueueWorker(new Worker(callback, priority, message));
```

Comment:
That's kind of unfortunate from a logging perspective.. @sam-github you might want to add a note about that to the README.md since it might be something people care about. Considering remote syslog is udp by default I'm guess this won't both most people anyway.